File: src/main/java/ru/market/pricing_service/repository/PriceListRepository.java
package ru.market.pricing_service.repository;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;
import ru.market.pricing_service.model.entity.PriceList;
import java.time.LocalDate;
import java.util.List;

@Repository
public interface PriceListRepository extends JpaRepository<PriceList, Integer> {
    List<PriceList> findByEffectiveDateLessThanEqualAndEndDateGreaterThanEqual(
            LocalDate date1, LocalDate date2);
}

File: src/main/java/ru/market/pricing_service/repository/PriceListRepository.kt
package ru.market.pricing_service.repository

import org.mapstruct.Named
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import ru.market.pricing_service.model.entity.PriceList

@Repository
interface PriceListRepository : JpaRepository<PriceList, Int> {
    @Named("priceListById")
    override fun getReferenceById(id: Int): PriceList
}